src/NHibernate/Logging.cs

```diff
@@ -316,118 +738,254 @@ static Log4NetLogger()
 		{
 			if (IsWarnEnabled)
-				WarnExceptionDelegate(logger, message, exception);
+				WarnExceptionDelegate(_logger, message, exception);
 		}
 
 		public void WarnFormat(string format, params object[] args)
 		{
 			if (IsWarnEnabled)
-				WarnFormatDelegate(logger, format, args);
+				WarnFormatDelegate(_logger, format, args);
+		}
+	}
+
+	/// <summary>
+	/// Extensions method for logging.
+	/// </summary>
+	public static class NHibernateLoggerExtensions
+	{
+		public static bool IsDebugEnabled(this INHibernateLogger logger) => logger.IsEnabled(InternalLogLevel.Debug);
+		public static bool IsInfoEnabled(this INHibernateLogger logger) => logger.IsEnabled(InternalLogLevel.Info);
+		public static bool IsWarnEnabled(this INHibernateLogger logger) => logger.IsEnabled(InternalLogLevel.Warn);
+		public static bool IsErrorEnabled(this INHibernateLogger logger) => logger.IsEnabled(InternalLogLevel.Error);
+		public static bool IsFatalEnabled(this INHibernateLogger logger) => logger.IsEnabled(InternalLogLevel.Fatal);
+
+		public static void Fatal(this INHibernateLogger logger, Exception exception, string format, params object[] args)
+		{
+			logger.Log(InternalLogLevel.Fatal, new InternalLogValues(format, args), exception);
 		}
+
+		public static void Fatal(this INHibernateLogger logger, string format, params object[] args)
+		{
+			logger.Log(InternalLogLevel.Fatal, new InternalLogValues(format, args), null);
+		}
+
+		public static void Error(this INHibernateLogger logger, Exception exception, string format, params object[] args)
+		{
+			logger.Log(InternalLogLevel.Error, new InternalLogValues(format, args), exception);
+		}
+
+		public static void Error(this INHibernateLogger logger, string format, params object[] args)
+		{
+			logger.Log(InternalLogLevel.Error, new InternalLogValues(format, args), null);
+		}
+
+		public static void Warn(this INHibernateLogger logger, Exception exception, string format, params object[] args)
+		{
+			logger.Log(InternalLogLevel.Warn, new InternalLogValues(format, args), exception);
+		}
+
+		public static void Warn(this INHibernateLogger logger, string format, params object[] args)
+		{
+			logger.Log(InternalLogLevel.Warn, new InternalLogValues(format, args), null);
+		}
+
+		public static void Info(this INHibernateLogger logger, Exception exception, string format, params object[] args)
+		{
+			logger.Log(InternalLogLevel.Info, new InternalLogValues(format, args), exception);
+		}
+
+		public static void Info(this INHibernateLogger logger, string format, params object[] args)
+		{
+			logger.Log(InternalLogLevel.Info, new InternalLogValues(format, args), null);
+		}
+
+		public static void Debug(this INHibernateLogger logger, Exception exception, string format, params object[] args)
+		{
+			logger.Log(InternalLogLevel.Debug, new InternalLogValues(format, args), exception);
+		}
+
+		public static void Debug(this INHibernateLogger logger, string format, params object[] args)
```

Comment:
There are 229 calls to this method with no arguments. but the array will be allocated anyway.
401 calls with 1 argument
88 calls with 2 arguments
And 32 calls with 3 or more arguments